modules/shared/src/main/scala/org/tessellation/storage/LocalFileSystemStorage.scala

```diff
@@ -36,38 +38,40 @@ abstract class LocalFileSystemStorage[F[_]: KryoSerializer, A <: AnyRef: ClassTa
     F.delay { (a / fileName).exists }
   }
 
-  def read(fileName: String): EitherT[F, Throwable, A] = readBytes(fileName).flatMap { bytes =>
-    EitherT.fromEither { KryoSerializer[F].deserialize[A](bytes) }
-  }
+  def read(fileName: String): F[Option[A]] =
+    readBytes(fileName)
+      .map(_.flatMap { bytes =>
+        KryoSerializer[F].deserialize[A](bytes).toOption
```

Comment:
So we won't be able to tell whether the file was not found or bytes couldn't be deserialized? Maybe we could raise the `Either` to `F`, to get the error whenever some error occurred and only get `None` when file wasn't there and `Some` when it was there and got deserialized successfully.